FoodApp/Controllers/RequestsController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using FoodApp.Data;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FoodApp.Controllers
+{
+    [Authorize(Roles = "Administrator")]
+    public class RequestsController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+
+        public RequestsController(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.Restaurant.Where(r => r.State=="Pending") .ToListAsync());
+        }
+
+        public async Task<IActionResult> Details(string id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
```

Comment:
We could use this instead for more readability.
`if (id is null) return NotFound();`

Reply:
No :)